src/dune_engine/build.ml

```diff
@@ -167,7 +167,7 @@ let memoize name t = Memo { name; id = Type_eq.Id.create (); t }
 let source_tree ~dir =
   let dep_set = Dep.Set.source_tree dir in
   let+ () = deps dep_set in
-  Dep.Set.paths dep_set ~eval_pred:(fun _ -> Path.Set.empty)
```

Comment:
Why is it OK to change `eval_pred` here?

Reply:
`Dep.Set.source_tree` returns a set of:

* File dependencies from source files
* Empty globs for directories

Neither of those are changed by setting `eval_pred`. This is why we got a bit lazy and just used a dummy value here.